clients/src/test/java/org/apache/kafka/common/metrics/MetricsTest.java

```diff
@@ -212,6 +213,18 @@ public void testQuotas() {
     }
 
     @Test
+    public void testQuotasEquality() {
+        final Quota quota1 = Quota.lessThan(10.5);
+        final Quota quota2 = Quota.moreThan(10.5);
+
+        assertFalse(quota1.equals(quota2));
+
+        Quota quota3 = Quota.moreThan(10.5);
```

Comment:
I suppose you can make this final as well since quota1 and 2 are final.
Another nit, can you add a message to the assertions? Otherwise, assertFalse will print a generic failure "Expected true received false" which doesn't really convey much information about the failing assertion.